### PR TITLE
Make Crafting Input Bus / Hatch compatible with some non-GPL multis

### PR DESCRIPTION
**Pull request by Jonodonozym**
This works by exposing the bus contents to `getStoredItems()` and `getStoredFluids()` methods. This makes the buses compatible with multis that override `checkProcessing()` with special logic and rely on the aforementioned methods.

Examples include Multi-Smelter and Industrial Arc Furnace.

### CODE REVIEW

File: src/main/java/gregtech/common/tileentities/machines/GT_MetaTileEntity_Hatch_CraftingInput_ME.java [hunk 1]
```diff
@@ -960,4 +960,38 @@ public boolean hasCustomName() {
...
+            }
+        }
+
+        return itemStacks;
+    }
+
+    public ArrayList<FluidStack> getCombinedFluidInputs() {
+        ArrayList<FluidStack> fluidStacks = new ArrayList<>();
+
+        if (!this.supportFluids) {
```
Comment (harrynull) [inline src/main/java/gregtech/common/tileentities/machines/GT_MetaTileEntity_Hatch_CraftingInput_ME.java:985]: This should not be necessary because no fluids could get into the internal inventory in the first place

**Comment by harrynull** [comment]
This would break the internal isolation between different patterns

File: src/main/java/gregtech/common/tileentities/machines/GT_MetaTileEntity_Hatch_CraftingInput_ME.java (hunk 1 above)
Comment (Jonodonozym) [reply src/main/java/gregtech/common/tileentities/machines/GT_MetaTileEntity_Hatch_CraftingInput_ME.java]: It saves time on iterating all inventories when we know they can't contain fluids.

File: src/main/java/gregtech/common/tileentities/machines/GT_MetaTileEntity_Hatch_CraftingInput_ME.java [hunk 2]
```diff
@@ -722,7 +722,7 @@ private void onPatternChange(int index, ItemStack newItem) {
         needPatternSync = true;
     }
 
-    private ItemStack[] getSharedItems() {
+    public ItemStack[] getSharedItems() {
```
Comment (harrynull) [inline src/main/java/gregtech/common/tileentities/machines/GT_MetaTileEntity_Hatch_CraftingInput_ME.java:725]: Why is this changed to `public`?

**Review by harrynull — APPROVED** [review]
lgtm with one nit

File: src/main/java/gregtech/common/tileentities/machines/GT_MetaTileEntity_Hatch_CraftingInput_ME.java (hunk 2 above)
Comment (Dream-Master) [reply src/main/java/gregtech/common/tileentities/machines/GT_MetaTileEntity_Hatch_CraftingInput_ME.java:725]: @Jonodonozym
